airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/PerStreamStateMessageTest.java

```diff
@@ -0,0 +1,61 @@
+package io.airbyte.integrations.standardtest.destination;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.base.FailureTrackingAirbyteMessageConsumer;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StreamDescriptor;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+public abstract class PerStreamStateMessageTest {
+  protected abstract Consumer<AirbyteMessage> getMockedConsumer();
+
+  protected abstract FailureTrackingAirbyteMessageConsumer getMessageConsumer();
+
+  @Test
+  void runATest() throws Exception {
```

Comment:
Nice! this is a super clear test

Reply:
oh I just realized that I forgot to rename it...